### ISSUE TITLE
limit of 600 PR ?

### ISSUE THREAD

**Issue by pborreli** [karanjthakkar/showmyprs.com#10]
https://showmyprs.com/user/pborreli indicates `Total pull requests sent: 600` 

It should be ~1204 instead

**Comment by shurcooL** [comment]
https://showmyprs.com/user/shurcooL says 722 PRs, so that means 600 cannot be an explicit limit. There may be another issue causing not all PRs to be displayed though.

Where did you get the 1204 number from?

**Comment by pborreli** [comment]
from https://github.com/pulls 

![image](https://cloud.githubusercontent.com/assets/77759/23527185/d9b8a3b4-ff8c-11e6-8ec7-f579f6dc1c10.png)

**Comment by shurcooL** [comment]
Got it. For me, it lines up pretty well. GitHub shows 7 open + 718 closed PRs created, which adds up to 725. showmyprs gives me 722, so there is a discrepancy of 3 PRs that I can't explain.

![image](https://cloud.githubusercontent.com/assets/1924134/23527903/bf6a2fd6-ff65-11e6-9983-e07882f94fce.png)

**Comment by shurcooL** [comment]
Here's an idea, are any of your PRs made in private repositories? showmyprs wouldn't be able to show those, since it uses public data only.

**Comment by shurcooL** [comment]
Yeah, I think that's exactly it. If I select "Public repositories only" under Repository visibility" filter, then GitHub shows me exactly 722 PRs in total (3 are in private repos).

![image](https://cloud.githubusercontent.com/assets/1924134/23527980/163ceaa6-ff66-11e6-997e-80304f62d2a8.png)

Can you try the same and see what you get?

**Comment by pborreli** [comment]
![image](https://cloud.githubusercontent.com/assets/77759/23528057/4dd712a0-ff90-11e6-9b2e-ca35b44115b6.png)

**Comment by shurcooL** [comment]
I see, then there's definitely some sort of issue in showmyprs.

I can confirm that https://github.com/search?q=type%3Apr+author%3Apborreli+is%3Apublic&type=Issues finds 1210 public PRs for your username:

![image](https://cloud.githubusercontent.com/assets/1924134/23528300/59a88a56-ff67-11e6-9883-f7ac11cb89a6.png)

I'll leave this to @karanjthakkar to investigate further.

**Comment by pborreli** [comment]
thanks @shurcooL for the help 💯👍

**Comment by karanjthakkar** [comment]
@pborreli Sorry for getting back to you so late about this. I have been travelling thee past few days.

1. The _probable_ reason why the first time you were shown a total of only 600, was because I'm using my personal access token for Github PR search and it has a limit of 30 for a window of 15 minutes with a max per page results of 100. So it is possible that when I started fetching list of PR's for your username, I had already consumed 24 searches for that window. 

2. Having refreshed your profile on my end, I found that it is now showing a total of 1000 PR's sent. This is a limit on Github's end. Since you have a total of 1200 something PR's I have to search 13 times with different page numbers. On the [11th page](https://api.github.com/search/issues?q=type:pr+author:pborreli&page=11&per_page=100), Github says: `Only the first 1000 search results are available`

In case 1, I am planning to fix by giving the user an option to login with their account so that I can use their access token for it. This should fix the incomplete PR count issue.

In case 2, however, there is no way for me to bypass the 1000 search results hard limit set by Github. I will investigate more on whether there are other ways to fetch your PR's without querying the search API.

Sorry if I can't fix this sooner but I hope this gives you more clarity on what the issue is.

cc: @shurcooL Just FYI :) Thanks for chipping in before and helping investigate this with @pborreli 💯

**Comment by pborreli** [comment]
wow thanks for the investigation, 

I would log without problem is the scope is readonly and public, 

Maybe the solution without the need of login would be the search the api looping on year ? like in this [answer](http://stackoverflow.com/questions/37602893/github-search-limit-results#comment70552977_37639739) ?

**Comment by karanjthakkar** [comment]
@pborreli Thanks for searching for a workaround this issue. I think this is doable. I might need to refactor some logic of recursively fetching newer search pages. I'll add this to my list of todos. Alternatively, if you would like to take a shot at a PR for this, I'd be more than happy to help you out with that!
